### ISSUE TITLE
[Feature Request] Implement Degree Planner Functionality

### ISSUE THREAD

**Issue by ZzRanger** [UTDNebula/planner#65]
# Overview

The Degree Planner tool inside Nebula Planner is what students will use to plan out and customize the college experience they want to have. Currently, the Degree Planner has been partially implemented, but some things still need to be done before it is ready to be integrated into the rest of the project.

# Proposal

1. Create the ability to add courses 
2. Develop a method for users to search for UTD courses not part of their degree plan
3. Allow users to add/remove semesters 
4. Build a new plan creation flow

# Supplementary Information

The design of the Degree Planner will aim to follow mockups from the "Nebula Mockups" Figma file.
